src/doc/trpl/unsafe.md

```diff
@@ -127,6 +110,111 @@ The latter assumption allows the compiler to optimize more
 effectively. As can be seen, actually *creating* a raw pointer is not
 unsafe, and neither is converting to an integer.
 
+## Reference and pointer aliasing
+
+Several examples can give a better comprehension of the reference and pointer
+aliasing rules.
+
+First of all, using `unsafe` code to circumvent the restrictions on references
+leads to undefined behaviour. For example, the following code creates two
+aliasing `&mut` references, and is therefore illegal.
```

Comment:
I'm not sure why I had an old branch set to my current Rust HEAD, but the code is still there there in 6d8342f5:

```rust
impl<'b, T> DerefMut<T> for RefMut<'b, T> {
    #[inline]
    fn deref_mut<'a>(&'a mut self) -> &'a mut T {
        unsafe { &mut *self._parent.value.get() }
    }
}
```